sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -392,90 +392,87 @@ class FilesElement extends Polymer.Element {
    */
   async _upload() {
     const inputElement = <HTMLInputElement>this.$.altFileUpload;
-    const files = inputElement.files;
+    const files = [...<any>inputElement.files];
     const currentPath = this.currentPath;
     const uploadPromises: Promise<any>[] = [];
 
-    if (files) {
-      // Find out if there's at least one large file.
-      let hasLargeFile = false;
-      for (let i = 0; i < files.length; ++i) {
-        if (files[i].size > this._uploadFileSizeWarningLimit) {
-          hasLargeFile = true;
-          break;
-        }
-      }
+    // Find out if there's at least one large file.
+    const hasLargeFile = files.some((file: File) => 
+        file.size > this._uploadFileSizeWarningLimit);
+
+    let proceedWithUpload = true;
+    // If there's at least one large file, show a dialog to confirm the user
+    // wants to continue with the upload.
+    if (hasLargeFile) {
+      let warningMsg = files.length > 1 ? 'Some of the files you selected are '
+                                          : 'The file you selected is ';
+      warningMsg += 'larger than 25MB. You might experience browser freeze or crash';
+      const dialogOptions: DialogOptions = {
+        title: 'Warning: Large File',
+        messageHtml: warningMsg,
+        okLabel: 'Upload Anyway',
+      };
 
-      let proceedWithUpload = true;
-      // If there's at least one large file, show a dialog to confirm the user
-      // wants to continue with the upload.
-      if (hasLargeFile) {
-        let warningMsg = files.length > 1 ? 'Some of the files you selected are '
-                                            : 'The file you selected is ';
-        warningMsg += 'larger than 25MB. You might experience browser freeze or crash';
-        const dialogOptions: DialogOptions = {
-          title: 'Warning: Large File',
-          messageHtml: warningMsg,
-          okLabel: 'Upload Anyway',
-        };
-
-        const result: BaseDialogCloseResult = 
-          await Utils.showDialog(BaseDialogElement, dialogOptions);
-          proceedWithUpload = result.confirmed === true;
-      }
+      const result: BaseDialogCloseResult = 
```

Comment:
I'd like to keep it for clarity. Even thought `showDialog` returns a `BaseDialogCloseResult`, this is a base type, and the actual type returned will differ according to the dialog type passed. So it's just a coincidence that we really are expecting a `BaseDialogCloseResult` in this case, and I think it's worth being explicit about that.